lib/gulp-index-file-stream.js

```diff
@@ -34,7 +34,10 @@ IndexFileStream.prototype._read = function() {
 
   this._indexStatic.generate(url, {
     meta: this._options.meta || {}
-  }, function(html) {
+  }, function(err, html) {
+    if (err) {
+      return this.emit('error', err);
+    }
```

Comment:
Ah, yeah, it is to preserve the ordering. Instead of using `.pop()` in the `_read` method I should've just used [`.shift()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift), but because I constantly get confused between shift and unshift and didn't want to look up the docs, at the time I just opted to reverse the array being popped-from instead! :sweat_smile: